File: backend/michicondrias_carnet/app/schemas/carnet.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class PrescriptionBase(BaseModel):
    medication_name: str
    dosage: str
    frequency_hours: int
    duration_days: int
    instructions: Optional[str] = None

class PrescriptionCreate(PrescriptionBase):
    pass

class PrescriptionResponse(PrescriptionBase):
    id: str
    medical_record_id: str

    class Config:
        from_attributes = True

class MedicationReminderBase(BaseModel):
    prescription_id: str
    pet_id: str
    remind_at: str
    sent: bool = False

class MedicationReminderResponse(MedicationReminderBase):
    id: str

    class Config:
        from_attributes = True

class MedicalRecordBase(BaseModel):
    pet_id: str
    veterinarian_id: Optional[str] = None
    clinic_id: Optional[str] = None
    appointment_id: Optional[str] = None
    reason_for_visit: str
    diagnosis: Optional[str] = None
    treatment: Optional[str] = None
    weight_kg: Optional[float] = None
    temperature_c: Optional[float] = None
    notes: Optional[str] = None

class MedicalRecordCreate(MedicalRecordBase):
    prescriptions: Optional[list[PrescriptionCreate]] = []

class MedicalRecordUpdate(BaseModel):
    reason_for_visit: Optional[str] = None
    diagnosis: Optional[str] = None
    treatment: Optional[str] = None
    weight_kg: Optional[float] = None
    notes: Optional[str] = None

class MedicalRecordResponse(MedicalRecordBase):
    id: str
    date: datetime
    prescriptions: list[PrescriptionResponse] = []

    class Config:
        from_attributes = True

class VaccineBase(BaseModel):
    pet_id: str
    name: str
    next_due_date: Optional[datetime] = None
    administered_by_vet_id: Optional[str] = None
    batch_number: Optional[str] = None
    notes: Optional[str] = None

class VaccineCreate(VaccineBase):
    pass

class VaccineUpdate(BaseModel):
    name: Optional[str] = None
    next_due_date: Optional[datetime] = None
    batch_number: Optional[str] = None
    notes: Optional[str] = None

class VaccineResponse(VaccineBase):
    id: str
    date_administered: datetime

    class Config:
        from_attributes = True

File: frontend/src/lib/services/carnet.ts
import { apiFetch } from "../api";

export interface Prescription {
    id: string;
    medical_record_id: string;
    medication_name: string;
    dosage: string;
    frequency_hours: number;
    duration_days: number;
    instructions: string | null;
}

export interface MedicalRecordCreate {
    pet_id: string;
    reason_for_visit: string;
    diagnosis?: string;
    treatment?: string;
    weight_kg?: number;
    temperature_c?: number;
    notes?: string;
    appointment_id?: string;
    prescriptions: {
        medication_name: string;
        dosage: string;
        frequency_hours: number;
        duration_days: number;
        instructions?: string;
    }[];
}

export interface MedicalRecord {
    id: string;
    pet_id: string;
    veterinarian_id: string | null;
    clinic_id: string | null;
    appointment_id: string | null;
    date: string;
    reason_for_visit: string;
    diagnosis: string | null;
    treatment: string | null;
    weight_kg: number | null;
    temperature_c: number | null;
    notes: string | null;
    prescriptions: Prescription[];
}

export interface Vaccine {
    id: string;
    pet_id: string;
    name: string;
    date_administered: string;
    next_due_date: string | null;
    administered_by_vet_id: string | null;
    batch_number: string | null;
    notes: string | null;
}

export async function getRecordsByPet(petId: string): Promise<MedicalRecord[]> {
    return apiFetch<MedicalRecord[]>("carnet", `/records/pet/${petId}`);
}

export async function createRecord(record: MedicalRecordCreate): Promise<MedicalRecord> {
    return apiFetch<MedicalRecord>("carnet", "/records/", {
        method: "POST",
        body: JSON.stringify(record),
    });
}

export async function getVaccinesByPet(petId: string): Promise<Vaccine[]> {
    return apiFetch<Vaccine[]>("carnet", `/vaccines/pet/${petId}`);
}

export async function createVaccine(vaccine: Omit<Vaccine, "id" | "date_administered">): Promise<Vaccine> {
    return apiFetch<Vaccine>("carnet", "/vaccines/", {
        method: "POST",
        body: JSON.stringify(vaccine),
    });
}
